hearth/lib/hearth/interceptor.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+require_relative 'interceptor/context'
+require_relative 'interceptor/list'
```

Comment:
I don't think it makes sense to module-ize interceptors. I think like PluginList, this should be InterceptorList (flat) and also InterceptorContext.